### PR TITLE
vp9dec: fix regression failure, issue #347

### PR DESCRIPTION
**Pull request by clearlylin**
The issue results from the update of avformat.

### CODE REVIEW

File: tests/decodeinputavformat.cpp [hunk 1]
```diff
@@ -53,7 +59,11 @@ bool DecodeInputAvFormat::initInput(const char* fileName)
             if (ctx->extradata && ctx->extradata_size)
                 m_codecData.append((char*)ctx->extradata, ctx->extradata_size);
             m_videoId = i;
-            setResolution(ctx->coded_width, ctx->coded_height);
+            if (ctx->coded_width > width || ctx->height > height) {
```
Comment (lizhong1008) [inline tests/decodeinputavformat.cpp]: using "||" may set a larger width and a smaller height. Would better judge height and width independent.

File: tests/decodeinputavformat.cpp [hunk 2]
```diff
@@ -38,9 +38,15 @@ DecodeInputAvFormat::DecodeInputAvFormat()
 
 bool DecodeInputAvFormat::initInput(const char* fileName)
 {
+    uint16_t width = 0, height = 0;
     int ret = avformat_open_input(&m_format, fileName, NULL, NULL);
     if (ret)
         goto error;
+    if (m_format->nb_streams) {
+        AVCodecContext* ctx = m_format->streams[0]->codec;
```
Comment (xuguangxin) [inline tests/decodeinputavformat.cpp]: why we need this? streams[0] maybe a audio track

**Comment by xuguangxin** [comment]
please give some detailed description in your commit log. why we need this fix?

**Comment by clearlylin** [comment]
edited it

**Pull request closed without merging**